src/FrameHelper.h

```diff
@@ -2,171 +2,225 @@
-            /**
-             * Loads JPEG image from memory without decompressing it.
-             * If frame.size=={0,0} before calling this function, it is set automatically based on the JPEG header.
-             */
-            static void loadFrameJPEG(const uint8_t* src, size_t size, base::samples::frame::Frame &frame);
+#include <stddef.h>
+// #include <base/samples/compressed_frame.h>
+
+// TODO
+// at the moment some functions call copyImageIndependantAttributes
+// automatically
+// this should be replaced by a flag
+
+namespace frame_helper {
+    class FrameHelper {
+    private:
+        // the buffer are used to convert one frame to an other one
+        // if src and dst attributes are not changing no memory allocation
+        // is done
+        // DO NOT USE THIS VARIABLES INSIGHT any function other than convert
+        base::samples::frame::Frame frame_buffer;
+        base::samples::frame::Frame frame_buffer2;
+
+        // used insight convertColor
+        base::samples::frame::Frame frame_buffer3;
+
+        // mapping matrix for undistort
+        // if the size of the image is not changing
+        // the mapping is calculated only once
+        CameraCalibrationCv calibration;
+
+    public:
+        FrameHelper();
+
+        // converts one frame to another frame
+        // this can be used to convert colors resize and undistort frames
+        // the attributes are implied by the dst attributes
+        // call setCalibrationParameters to set the calibration parameters
+        // this is none static because an internal buffer is used which is not be resized
+        // if the src and dst have always the same attributes
+        void convert(const base::samples::frame::Frame& src,
+            base::samples::frame::Frame& dst,
+            int offset_x = 0,
+            int offset_y = 0,
+            ResizeAlgorithm const& resize_algorithm = INTER_LINEAR,
+            bool bundistort = false,
+            UndistortAlgorithm const& undistort_algorithm = UNDISTORT_CUBIC);
+
+        // sets the calibration paramter for the camera
+        // parameter. Be aware that this does not set the R and P parameter
+        // for stereo undistort. Use the method that takes a cameraCalibrationCv
+        // instead.
+        // para              intrinsic and distortion parameters
+        void setCalibrationParameter(const CameraCalibration& para);
+
+        // sets the calibration parameters for the camera
+        // which uses R and P matrices for stereo calibration as well
+        void setCalibrationParameter(const CameraCalibrationCv& para);
+
+        // this is a convenience function to undistort a frame
+        // call setCalibrationParameter first
+        // see cv::remap
+        void undistort(const base::samples::frame::Frame& src,
+            base::samples::frame::Frame& dst,
+            UndistortAlgorithm const& resize_algorithm);
```

Comment:
Should this be named `undistort_algorithm`? And I think it should have a default value for API compatibility.